src/test/java/com/commercetools/project/sync/product/ProductSyncerTest.java

```diff
@@ -155,102 +152,18 @@ void transform_WithAttributeReferences_ShouldReplaceProductReferenceIdsWithKeys(
     assertThat(testLogger.getAllLoggingEvents()).isEmpty();
   }
 
-  @Test
-  void
-      transform_WithIrresolvableAttributeReferences_ShouldSkipProductsWithIrresolvableReferences() {
```

Comment:
Why this test is not necessary anymore?

Reply:
Because it is not handled in Java-Sync. Previously it was handled in Project-sync with the warning loggers.

Reply:
But the scenario is valid right, we should not remove scenarios because it does not have the same assertion as before, what happens if you have an irresolvable attribute reference? I will check it to have more clear answer otherwise this is not what I am asking.

Reply:
I also figured out this is a unit test, which does not matter much... we might check if we have a similar integration test as mocking all calls that do not make sense to me much.

Reply:
> Note : We had warning logger for Unresolvable references, But now Library is no more handling this case to add warning logger for unresolvable references. So I have removed tests for this part. https://github.com/commercetools/commercetools-project-sync/pull/268/files#diff-4514648cfaf3d44c0ee256fc715c4e327733fd1933043c07b92e8f5e30d8c083L160 Let me know if you have any suggestion on this part?

This seems not a question to me as you are not explaining the background of this so I need to investigate. When I check the java-sync code, I saw that we are filtering those references if their id does not exist in the target project and simply there is no trace of why those products are filtered out because we are skipping mapping them and not returning the mapped draft.

This test case in java-sync below demonstrates it:

https://github.com/commercetools/commercetools-sync-java/blob/35c0df8928b0ecd68d8fc4c84d7e1da8d42fc747/src/test/java/com/commercetools/sync/products/service/impl/ProductTransformServiceImplTest.java#L200-L234

It passes 2 products to the mapper `toProductDrafts` and the method returns 1 product draft.
Also, the test cases in project-sync proved that.

Some problems I could point:
- It is not sth documented in Javadoc of the map method, which is unexpected behavior for the library user.
- No traceability of the mapping, this might cause some issues as some users could ask why those products are not synced and we would hard time reasoning it.

I could also understand the intent here, probably you could not find any logging mechanism in java-sync and decided to skip that, but that would be asked before doing a development so we could think of a mechanism/concept. 

In this case, I would suggest if the product can not be resolved you pass the product draft as is and in this case, it could be traced by logs when it fails on the later stage of sync. Also, I could see that mapper also throws an exception when the sync will fail because the attribute can not be resolved, but this case is skipped a bit hard to reason honestly.  Anyway, we need to work on it as the PR review is not enough, so a follow-up task and concept meeting would be nice to not suppress the issue.

For now, I'm not seeing this as a blocker for the release and I will resolve this discussion but would be nice to create an action for it as part of the current task.

Reply:
Thank you for your time on checking this. I have created a follow up ticket for the investigation and create a follow up tickets to resolve this part in both Sync-Java and Project-sync.